### PR TITLE
Add iOS fv build and network test template

### PR DESCRIPTION
**Pull request by ystefinko**
Test script have Simulator launch temporary disabled.
Enable -DOLP_SDK_ENABLE_TESTING=ON on iOS Azure psv job.
Remove obsolete travis script too. Minor names refactoring.
Add separate yml for FV scheduled job.
Add gitalab manual job.

Relates-To: OLPEDGE-1773, OLPDGE-2543

Signed-off-by: Yaroslav Stefinko <ext-yaroslav.stefinko@here.com>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
       shell: bash
 
   linux-gcc-psv-build-relwithdebinfo-no-cache:
-    name: Linux Build using gcc & tests & code coverage
+    name: Linux No-Cache Build using gcc RelWithDebInfo
```
Comment (andescu) [inline .github/workflows/ci.yml:37]:
```suggestion
    name: Linux gcc / OLP_SDK_ENABLE_DEFAULT_CACHE=OFF
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -47,3 +47,18 @@ jobs:
     - name: Compile project without cache
       run: ./scripts/linux/psv/build_psv_no_cache.sh
       shell: bash
+
+  ios-macos-10-fv-build-relwithdebinfo-test-network:
+    name: iOS macos-10 FV build & network test(disabled)
```
Comment (andescu) [inline .github/workflows/ci.yml:52]:
```suggestion
    name: iOS macos-10 FV build & network test
```

File: scripts/ios/fv/gh-actions-functional-network-test.sh [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+while [[ ${RC} -ne 0 ]];
+do
+        set +e
+        curl -s http://localhost:1080
+        RC=$?
+        sleep 0.2
+        set -e
+done
+
+export IP_ADDR=$(ipconfig getifaddr en0)
```
Comment (andrey-kashcheev) [inline scripts/ios/fv/gh-actions-functional-network-test.sh:56]: I thought that only Windows using `ipconfig`. Shouldn't it be `ifconfig`?

File: scripts/ios/fv/gh-actions-build-with-tests-fv.sh [hunk 4]
```diff
@@ -17,12 +17,18 @@
...
-    ..
-make -j
-cd ..
+cmake ../ -GXcode \
+    -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/iOS.cmake \
+    -DCMAKE_BUILD_TYPE=$BUILD_TYPE \
+    -DPLATFORM=iphoneos \
+    -DOLP_SDK_ENABLE_TESTING=ON \
+    -DSIMULATOR=YES \
+    -DOLP_SDK_BUILD_EXAMPLES=ON
```
Comment (andescu) [inline scripts/ios/fv/gh-actions-build-with-tests-fv.sh:32]: Why is this needed? Examples we only need to test ones.

File: scripts/ios/fv/gh-actions-functional-network-test.sh (hunk 3 above)
Comment (ystefinko) [reply scripts/ios/fv/gh-actions-functional-network-test.sh:56]: As you may check in job logs - it works on MacOS too :)

File: scripts/ios/fv/gh-actions-build-with-tests-fv.sh (hunk 4 above)
Comment (ystefinko) [reply scripts/ios/fv/gh-actions-build-with-tests-fv.sh]: thanks for finding it, fixed.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ystefinko) [reply .github/workflows/ci.yml]: thanks for finding it, fixed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ystefinko) [reply .github/workflows/ci.yml]: thanks for finding it, fixed.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -15,7 +15,7 @@ env:
 
 jobs:
   linux-gcc-psv-build-test-codecov:
-    name: Linux Build using gcc & tests & code coverage
+    name: Linux gcc build & tests & code coverage
```
Comment (andescu) [inline .github/workflows/ci.yml:18]: There doesn't seem to be any naming pattern here. I think we should find one. I propose to start first with the type of pipeline then with the OS and compiler, followed by additional steps, all separated by `/`.
E.g. `PSV / Linux gcc 5.4 / Tests / Code coverage`

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -34,7 +34,7 @@ jobs:
       shell: bash
 
   linux-gcc-psv-build-relwithdebinfo-no-cache:
-    name: Linux Build using gcc & tests & code coverage
+    name: Linux gcc / OLP_SDK_ENABLE_DEFAULT_CACHE=OFF
```
Comment (andescu) [inline .github/workflows/ci.yml:37]: Same here.

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -47,3 +47,18 @@ jobs:
     - name: Compile project without cache
       run: ./scripts/linux/psv/build_psv_no_cache.sh
       shell: bash
+
+  ios-macos-10-fv-build-relwithdebinfo-test-network:
+    name: iOS macos-10 FV build & network test
```
Comment (andescu) [inline .github/workflows/ci.yml:52]: Same here.

**Review by andescu — COMMENTED** [review]
What is the app.log and why do you commit it? These logs to be a dumbfile of the std::cout which should not be included in the source. If this is so, please remove.

File: mock-server-cert.pem [hunk 8]
```diff
@@ -0,0 +1,22 @@
+-----BEGIN CERTIFICATE-----
```
Comment (andescu) [inline mock-server-cert.pem:1]: This file doesn't belong here. Please move to `tests/utils/mock-server/`

Comment (ystefinko) [reply mock-server-cert.pem]: done, thanks

**Review by andescu — APPROVED** [review]

**Review by jmoro3 — APPROVED** [review]

**Review by andescu — CHANGES_REQUESTED** [review]
Remove the additional PSV iOS job.

**Review by mykhailo-kuchma — APPROVED** [review]

**Comment by ystefinko** [comment]
> Remove the additional PSV iOS job.

done

**Review by andescu — APPROVED** [review]
